src/platforms/esp32/components/avm_sys/include/esp32_sys.h

```diff
@@ -114,6 +124,8 @@ struct ESP32PlatformData
 #endif
     mbedtls_ctr_drbg_context random_ctx;
     bool random_is_initialized;
+
+    struct SyncList misc_entries;
```

Comment:
This not the most elegant solution maybe, but for sure avoids changing multiple times ESP32PlatformData.
I suggest keeping this for attaching structs that are not so frequently used, such as mount points, for which a linear search is good enough.

@pguyot  on a second thought `misc_entries` might be not the best name, what about `attachments_list`?